tests/test_spectrum_assignment.py

```diff
@@ -0,0 +1,378 @@
+
+@pytest.mark.parametrize('nval', [0, -300, 500])
+@pytest.mark.parametrize('mval', [1, 600])
+def test_wrong_values(nval, mval, setup):
+    """ test n or m not applicable values
+    """
+    network, oms_list = setup
+    test = oms_list[5].assign_spectrum(nval, mval)
+    print(f'n, m values should raise an error {nval}, {mval}')
+    expected = False
+    if (nval + mval) == 1:
+        expected = True
+    print(nval, mval)
+    assert test is expected
+
+def test_bitmap_assignment(setup):
+    """ test that a bitmap can be assigned
+    """
+    network, oms_list = setup
+
+    oms_list[5].assign_spectrum(13, 7)
+
+    btmp = deepcopy(oms_list[5].spectrum_bitmap.bitmap)
+    freq_min = 191300000000000.0
+    freq_max = 196100000000000.0
+    # try a first assignment that must pass
+    spectrum_btmp = Bitmap(freq_min, freq_max, grid=0.00625e12, guardband=0.15e12, bitmap=btmp)
+
+    # try a wrong asignment that should not pass
+    btmp = btmp[1:-1]
+    test = False
+    try:
+        spectrum_btmp = Bitmap(freq_min, freq_max, grid=0.00625e12, guardband=0.15e12, bitmap=btmp)
+    except SpectrumError:
+        test = True
+
+    print('bitmap direct assignment should create an error if length is not consistant with' +\
+          'provided values')
+    assert test
+
+@pytest.fixture()
+def data(eqpt):
+    """ common setup for service list: builds service only once
+    """
+    with open(SERVICE_FILENAME, encoding='utf-8') as my_f:
+        data = loads(my_f.read())
+    return data
+
+@pytest.fixture()
+def requests(eqpt, data):
+    """ common setup for requests, builds requests list only once
+    """
+    equipment = eqpt
+    rqs = requests_from_json(data, equipment)
+    return rqs
+
+def test_spectrum_assignment_on_path(eqpt, setup, requests):
+    """ test assignment functions on path and network
+    """
+    equipment = eqpt
+    network, oms_list = setup
+    rqs = requests
+    req = [rqs[1]]
+    pths = compute_path_dsjctn(network, equipment, req, [])
+
+    print(req)
+    for nval in range(100):
+        (center_n, startn, stopn), path_oms = spectrum_selection(pths[0], oms_list, 4)
+        pth_assign_spectrum(pths, req, oms_list, [find_reversed_path(pths[0])])
+        print(f'testing on following oms {path_oms}')
+        # check that only 96 channels are feasible
+        if nval >= 96:
+            print(center_n, startn, stopn)
+            print('only 96 channels of 4 slots pass in this grid')
+            assert center_n is None and startn is None and stopn is None
+        if nval < 96:
+            print(center_n, startn, stopn)
+            print('at least 96 channels of 4 slots should pass in this grid')
+            assert center_n is not None and startn is not None and stopn is not None
+        # reset req[0]: since pth_assign_spectrum adds N and M attribute to the class,
+        # removing N and M simulates a brand new request at each loop.
+        delattr(req[0],'N')
+        delattr(req[0],'M')
```

Comment:
I think that `@pytest.parametrize` is better -- the currently chosen approach requires an extra explanation

Reply:
That one was a bit more difficult to handle with parametrize, because the networks need to be loaded at each loop of nval, and a parametrize deletes previous loads. I used a deepcopy of the request instead to avoid delattr